countdown.rb

```diff
@@ -5,64 +5,115 @@
 
 require_relative 'helpers'
 
-clear
 
+a = Artii::Base.new :font => 'ogre'
+b = Artii::Base.new :font => 'catwalk'
+c = Artii::Base.new :font => 'cosmic'
+d = Artii::Base.new :font => 'invita'
+clear
 puts 'Introduce el tiempo en minutos'
 minutes = gets.chomp.to_i
 
 clear
 
 puts 'Introduce la clave de desactivación'
 bomb_key = gets.chomp
-
+accumulated_key = []
+(bomb_key.length).times { accumulated_key << "*" }
 bomb_deactivated = false
 
 final = Time.now + minutes * 60
 
 loop do
-  accumulated_key = []
+  
   pressed_key = nil  
+  count = 0
   begin
     clear
-    puts "SE ESTÁ ACABANDO EL TIEMPO..."
-    puts "#{format_time(time_remaining(final))}"
-
-    puts "Introduce la clave:\n\n"
-
-    print "#{accumulated_key.join}"
+    3.times { puts }
+    puts a.asciify("SE ESTA ACABANDO EL TIEMPO...")
+    3.times { puts }
+    puts d.asciify("#{format_time(time_remaining(final))}")
+    3.times { puts }
+    puts c.asciify("Introduce   la     clave   :\n\n")
+    3.times { puts }
+    print b.asciify("#{accumulated_key.join}")
+    break if Time.now > final
 
     Timeout.timeout(1) do
       pressed_key = STDIN.getch
       raise Timeout::Error
-    end
-
-    break if Time.now > final
+    end 
 
-  rescue Timeout::Error
-    if pressed_key
-        accumulated_key << pressed_key
+    rescue Timeout::Error
+      if pressed_key
+        accumulated_key[count] = pressed_key
+        count = count.next
         pressed_key = nil
-    end
-
-    if accumulated_key.join == bomb_key
+      end
+      
+      if accumulated_key.join == bomb_key
         bomb_deactivated = true
         break
-    end
-    
+      end
+      
+      if accumulated_key[bomb_key.length-1] != "*" && accumulated_key.join != bomb_key
+        final = final-30
+        accumulated_key.clear
+        (bomb_key.length).times { accumulated_key << "*" }
+        count = 0
+        pressed_key = nil
+      end
+      
     retry
   end
 end
 
 clear
 
 if bomb_deactivated
-  puts "Bomba desactivada! Ha sobrado #{format_time(time_remaining(final))}"
+  puts a.asciify("Bomba desactivada! Ha sobrado #{format_time(time_remaining(final))}")
```

Comment:
I love these two ASCII arts, but I would put them in a separate file the same way we do with helper functions.